x-pack/plugins/observability_solution/synthetics/server/routes/overview_trends/overview_trends.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { withSpan } from '@kbn/apm-utils';
+import { schema } from '@kbn/config-schema';
+import { SYNTHETICS_API_URLS } from '../../../common/constants';
+import { SyntheticsRestApiRouteFactory } from '../types';
+import { fetchTrends } from './fetch_trends';
+import { TrendKey } from '../../../public/apps/synthetics/state';
+
+export const createOverviewTrendsRoute: SyntheticsRestApiRouteFactory = () => ({
+  method: 'POST',
+  path: SYNTHETICS_API_URLS.OVERVIEW_TRENDS,
+  validate: {
+    body: schema.any(),
+  },
+  handler: async (routeContext): Promise<any> => {
+    return withSpan('fetch trends', async () => {
+      const esClient = routeContext.syntheticsEsClient;
+      const body = routeContext.request.body as TrendKey[];
+      const configs = body.reduce((acc: Record<string, string[]>, { configId, locationId }) => {
+        if (!acc[configId]) {
+          acc[configId] = [locationId];
+        } else {
+          acc[configId].push(locationId);
+        }
+        return acc;
+      }, {});
+      const results = await Promise.all(
+        Object.keys(configs).map((key) => fetchTrends(key, configs[key], esClient))
+      );
```

Comment:
instead of doing Promise.all we can use `esClient.msearch`  to perform multiple es queries at the same time.

Reply:
See https://github.com/elastic/kibana/pull/191747, this PR leverages that commit as well. We can make changes upstream there if there's anything implementation-wise you'd like done differently and then I'll resolve any conflicts on this branch.

Reply:
This was a really good recommendation, seeing really significant perf gains and can raise the chunk size now.